### PR TITLE
82 feature admin account creationblocking be

### PR DESCRIPTION
**Pull request by wabwirevalentine**
# Pull Request Template

## Description

This PR implements comprehensive user management functionality with role-based access control. The system allows administrators and super admins to manage users through a RESTful API with proper validation, security, and email notifications.

Fixes #(issue-number-if-applicable)

---

## Changes Made

- **User Management API Endpoints**:
  - `GET /api/user-management` - List users with role-based filtering
  - `POST /api/user-management` - Create new users with invitation emails
  - `GET /api/user-management/<user_id>` - Get specific user details
  - `PATCH /api/user-management/<user_id>` - Update user information
  - `DELETE /api/user-management/<user_id>` - Delete users
  - `PATCH /api/user-management/<user_id>/status` - Activate/deactivate users

- **Role-Based Access Control**:
  - Super Admin: Full access to all users including other admins
  - Admin: Can manage clients but cannot modify other admins or super admins
  - Client: No access to user management endpoints

- **Validation & Security**:
  - Comprehensive user data validation (email, phone, name, role)
  - Password generation with security requirements
  - Duplicate email/phone prevention
  - JWT authentication with role verification
  - Prevention of self-deletion and unauthorized modifications

- **Email Integration**:
  - Automated invitation emails with verification links
  - Background email sending using threading
  - Password inclusion in invitation emails

- **Testing**:
  - 20+ comprehensive test cases covering all scenarios
  - Role-based permission testing
  - Validation error testing
  - Edge case handling

---

### Automated Tests
- [x] All 20+ test cases passing
- [x] Role-based access control verified
- [x] Validation errors properly handled
- [x] Email mocking working correctly
- [x] Database integrity maintained

### Manual QA Verification
**Test Scenarios Covered:**
- [x] Admin can create/list/update/delete clients
- [x] Super admin can manage all users including admins
- [x] Clients cannot access user management endpoints
- [x] Proper error messages for validation failures
- [x] Duplicate email/phone prevention
- [x] Self-deletion prevention
- [x] Email invitation system

## Summary by CodeRabbit

* **New Features**
  * Role-based user management API (list, create, update, delete, change status) with invitation emails, temporary passwords and 24‑hour activation links.
  * Password strength validation utility.

* **Bug Fixes**
  * CORS origins configurable via environment, applied to API and verification routes (credentials preserved).
  * Verification errors now return HTTP 400 for invalid token purpose; added runtime logging for verification failures.

* **Tests**
  * Comprehensive user management test coverage.

* **Chores**
  * Default frontend verification URL updated to http://localhost:5177.

### CODE REVIEW

**Comment by Kimanthif** [comment]
Okay, though resolve conflicts in __init__.py

File: server/utils/password_validation.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
+def _is_valid_password(password: str) -> bool:
```
Comment (jkininga) [inline server/utils/password_validation.py:1]: I have also moved this function from user.py to here utils. I also remove the underscore before is .. and used it as is_valid_password, could you make that change so we dont clash?

**Review by jkininga — APPROVED** [review]
